### PR TITLE
Encryption v2 perf updates

### PR DESCRIPTION
**Pull request by rickle-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java [hunk 1]
```diff
@@ -50,6 +64,39 @@ public BlobTestBase(TOptions options) {
 
         blockBlobClient = blobClient.getBlockBlobClient();
         blockBlobAsyncClient = blobAsyncClient.getBlockBlobAsyncClient();
+
+        this.createBlob = createBlob;
+        this.singletonBlob = singletonBlob;
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java:69]: for the sake of simplicity I'd just remove `singletonBlob` and always create a blob per test.
For downloads it doesn't matter much and it' just making the code more complicated.

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java (hunk 1 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java:69]: @mikeharder specifically called out that download tests should only upload one blob for the whole run and all parallel instances should target downloading the same blob. Mike, do you have input here?

Comment (mikeharder) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java:69]: I don't feel too strongly either way.  Uploading a single blob makes setup a little faster (especially for `1GB parallel=8`), but it also makes the code a little more complex.  I do think it's nice if this is consistent across languages though, and the other languages currently upload a single blob.

Comment (kasobol-msft) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java]: btw. creating abstract base for upload and downloads should give flexibility to preserve the behavior without much complexity.

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java [hunk 2]
```diff
@@ -17,29 +16,42 @@
...
 
 public abstract class BlobTestBase<TOptions extends BlobPerfStressOptions> extends ContainerTest<TOptions> {
 
     public static final int DEFAULT_BUFFER_SIZE = 8192;
+    protected static  final String BLOB_NAME_PREFIX = "randomblobtest-";
     protected final BlobClient blobClient;
     protected final BlockBlobClient blockBlobClient;
     protected final BlobAsyncClient blobAsyncClient;
     protected final BlockBlobAsyncClient blockBlobAsyncClient;
+    private static final FakeKey kek;
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java:29]: nit: can we name it fakeEncryptionKey (or whatever name is good) ?

**Pull request merged**